### PR TITLE
migrate from drone ci to gh actions

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

PR migrations CI from drone to gh actions
**Solution:**

**Related Issue:**
https://github.com/harvester/harvester/issues/5327
**Test plan:**

### CODE REVIEW

File: scripts/validate [hunk 1]
```diff
@@ -15,5 +15,5 @@ if [[ -z "$(command -v golangci-lint)" ]]; then
 fi
 
 echo "Running: golangci-lint run"
-golangci-lint run
+golangci-lint run --timeout 10m
```
Comment (WebberHuang1118) [inline scripts/validate:18]: Hi @ibrokethecloud,
Can you help to explain why we need to introduce this timeout? Thanks.

Comment (ibrokethecloud) [reply scripts/validate:18]: the default timeout is 1m, and I noticed random failures due to golangci-lint timeout. This is likely due to the lack of resources on the runners.

**Review by Yu-Jack — COMMENTED** [review]
Same as others, we should generate amd64 and arm64 build, and move specified binary in Dockerfile.

File: .github/workflows/template-build.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          provenance: false
+          context: .
+          platforms: linux/amd64
+          load: true
+          file: package/Dockerfile
+          push: ${{ inputs.push }}
+          tags: ${{env.imageName}}:${{ inputs.release-tag-name }}
+          build-args: |
+            VERSION=${{ github.ref_name }}-${{ github.sha }}-head
```
Comment (Yu-Jack) [inline .github/workflows/template-build.yml:58]: BTW, this one isn't used in Dockerfile as well, so we could remove it.

Comment (ibrokethecloud) [reply .github/workflows/template-build.yml:58]: 👍

**Review by Yu-Jack — APPROVED** [review]
LGTM, thanks.

**Review by WebberHuang1118 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**